util/src/main/java/io/grpc/util/MultiChildLoadBalancer.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.util;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.grpc.ConnectivityState.CONNECTING;
+import static io.grpc.ConnectivityState.IDLE;
+import static io.grpc.ConnectivityState.READY;
+import static io.grpc.ConnectivityState.TRANSIENT_FAILURE;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.grpc.ConnectivityState;
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancerProvider;
+import io.grpc.Status;
+import io.grpc.SynchronizationContext;
+import io.grpc.SynchronizationContext.ScheduledHandle;
+import io.grpc.internal.ServiceConfigUtil.PolicySelection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/**
+ * The top-level load balancing policy.
+ */
+public abstract class MultiChildLoadBalancer extends LoadBalancer {
+
+  @VisibleForTesting
+  public static final int DELAYED_CHILD_DELETION_TIME_MINUTES = 15;
+  private static final Logger logger = Logger.getLogger(MultiChildLoadBalancer.class.getName());
+  private final Map<Object, ChildLbState> childLbStates = new HashMap<>();
+  private final Helper helper;
+  protected final SynchronizationContext syncContext;
+  private final ScheduledExecutorService timeService;
+  // Set to true if currently in the process of handling resolved addresses.
+  private boolean resolvingAddresses;
+
+  protected MultiChildLoadBalancer(Helper helper) {
+    this.helper = checkNotNull(helper, "helper");
+    this.syncContext = checkNotNull(helper.getSynchronizationContext(), "syncContext");
+    this.timeService = checkNotNull(helper.getScheduledExecutorService(), "timeService");
+    logger.log(Level.FINE, "Created");
+  }
+
+  protected abstract SubchannelPicker getInitialPicker();
+
```

Comment:
For initial and error picker, I guess it might be better to provide default. It can be a utility class here. Because I imagine most LBs have the same empty picker an error picker.

Reply:
Moved both error and empty pickers from Xds specific class to LoadBalancer and made them the defaults.